### ISSUE TITLE
tests fail on system with non-english culture

### ISSUE THREAD

**Issue by lg2de** [SonarSource/sonar-dotnet#3055]
### Description

Getting lastest master version the unit tests does not complete successfully on my PC.
They fail due to different culture settings (German).

### Repro steps

1. Switch (Windows) language to different language, e.g. German.
2. Compile the solution and run the tests.

### Expected behavior

All tests are run fine on every (Windows) system.

### Actual behavior

Some tests fail. Here is one example:

```
Test Name:	BuildError
Test FullName:	SonarAnalyzer.UnitTest.SonarAnalyzer.UnitTest.TestFramework.DiagnosticVerifierTest.BuildError
Test Source:	[...]\sonaranalyzer-dotnet\tests\SonarAnalyzer.UnitTest\TestFramework\DiagnosticVerifierTest.cs : line 178
Test Outcome:	Failed
Test Duration:	0:00:00

Test Name:	BuildError
Test Outcome:	Failed
Result StackTrace:	
bei FluentAssertions.Execution.LateBoundTestFramework.Throw(String message)
   bei FluentAssertions.Execution.TestFrameworkProvider.Throw(String message)
   bei FluentAssertions.Execution.CollectingAssertionStrategy.ThrowIfAny(IDictionary`2 context)
   bei FluentAssertions.Execution.AssertionScope.Dispose()
   bei FluentAssertions.Specialized.ExceptionAssertions`1.ExceptionMessageAssertion.Execute(IEnumerable`1 messages, String expectation, String because, Object[] becauseArgs)
   bei FluentAssertions.Specialized.ExceptionAssertions`1.WithMessage(String expectedWildcardPattern, String because, Object[] becauseArgs)
   bei SonarAnalyzer.UnitTest.TestFramework.DiagnosticVerifierTest.BuildError() in [...]\SonarAnalyzer.UnitTest\TestFramework\DiagnosticVerifierTest.cs:Zeile 188.
Result Message:	
Expected exception message to match the equivalent of 
"Unexpected build error [CS1514]: { expected on line 2", but 
"Unexpected build error [CS1514]: { erwartet. on line 2" does not.
Result StandardOutput:	
Actual diagnostics: 3
  Id: CS1514, Line: 2, [46, 46]
  Id: CS1513, Line: 2, [46, 46]
  Id: CS1022, Line: 5, [0, 1]
```

Currently the following tests fail:
*CheckFileLicenseTest.CheckFileLicense_WhenProvidingAnInvalidRegex_ShouldThrowException_CS
*MethodsShouldNotHaveTooManyLinesTest.MethodsShouldNotHaveTooManyLines_InvalidMaxThreshold_CS
*MethodsShouldNotHaveTooManyLinesTest.MethodsShouldNotHaveTooManyLines_InvalidMaxThreshold_VB
*DiagnosticVerifierTest.BuildError

### Known workarounds

Switch (Windows) language to English (US).

### Related information

* SonarC# Version:  8.2++ (latest)
* Visual Studio Version: 16.4.3

### Proposed solution

The tests should not validate text coming from operating system
I would like to adopt these tests.
Do you agree?

Another solution would be run the tests within wrapper changing current culture explicitly to en-us.

**Comment by andrei-epure-sonarsource** [comment]
thanks, @lg2de 

I agree that tests should not fail depending on the culture.

In the same time, these tests are not coming from the operating system, but from the Roslyn compiler framework.
- the tests that verify a warning message `AD0001` - this is the generic Roslyn analyzer error code - when Roslyn catches an exception thrown by one of the analyzers - so inside it, the text of the exception is thrown by our code. I expect only the `error` in `error AD0001` to be culture-specific (e.g. "Erreur" in French)...
- the test in `DiagnosticVerifierTest` verifies that our unit test framework will print out correctly the build error message (e.g. `CS1514`) - we sometimes check how our code behaves when the code does not compile, as our analyzer runs in the IDE while devs write their code, as well

To sum up, we want to validate the text coming from the compiler (our analyzer is just a plugin to the Roslyn compiler).

And, in the same time, if you want to modify these tests to keep asserting the compiler messages without failing on OS with a non-English culture settings, we will appreciate your contribution 😄
